src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp

```diff
@@ -33,11 +33,40 @@
 namespace miopen {
 namespace solver {
 
-static inline const std::vector<PerformanceImplicitGemmV4R1Dynamic>&
+struct TunableImplicitGemmV4R1Dynamic
+{
+    int BPerBlock;
+    int KPerBlock;
+    int EPerBlock;
+
+    int GemmNRepeat;
+
+    int GemmMPerThreadSubC;
+    int GemmNPerThreadSubC;
+
+    int GemmMLevel0Cluster;
+    int GemmNLevel0Cluster;
+    int GemmMLevel1Cluster;
+    int GemmNLevel1Cluster;
+
+    int InBlockCopyClusterLengths_E;
+    int InBlockCopyClusterLengths_N1;
+    int InBlockCopyClusterLengths_B;
+    int InBlockCopyClusterLengths_N2;
+
+    int WeiBlockCopyClusterLengths_E;
+    int WeiBlockCopyClusterLengths_K;
+
+    TunableImplicitGemmV4R1Dynamic(
+        int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int);
+    bool IsValid(const ConvolutionContext& ctx) const;
+};
+
+static inline const std::vector<TunableImplicitGemmV4R1Dynamic>&
 GetImplicitGemmV4R1DynamicTunables()
 {
     // clang-format off
-    static const std::vector<PerformanceImplicitGemmV4R1Dynamic> tunables = {
+    static const std::vector<TunableImplicitGemmV4R1Dynamic> tunables = {
```

Comment:
Please add a comment saying that the order of elements is important for performance.